osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs

```diff
@@ -95,6 +101,8 @@ public override void ApplyBeatmap(IBeatmap beatmap)
             )));
 
             targetMinimumHealth = BeatmapDifficulty.DifficultyRange(beatmap.BeatmapInfo.BaseDifficulty.DrainRate, min_health_target, mid_health_target, max_health_target);
+            targetMinimumHealth += drainLenience * (1 - targetMinimumHealth);
+            targetMinimumHealth = Math.Min(1, targetMinimumHealth);
```

Comment:
i don't quite follow this math.

set drainLenience = 1

targetMinimumHealth is calculated to be 0.2.

`targetMinimumHealth = 0.2 + 1 * (1 - 0.2)`

targetMinimumHealth = 1

wouldn't this make the drain higher?

Reply:
`targetMinimumHealth = 1` means the minimum health is 100%, so there's effectively no drain.

Reply:
so can this effectively be written as:

```csharp
var maximumDrop = BeatmapDifficulty.DifficultyRange(beatmap.BeatmapInfo.BaseDifficulty.DrainRate, min_health_target, mid_health_target, max_health_target);

targetMinimumHealth = 1 - Math.Clamp(maximumDrop * (1 - drainLenience), 0, 1);
```

Because I would prefer that (feels much easier to read).

Reply:
So plug in 0.3 (`max_health_target`) with a lenience of 0 (default):

```
tMH = 1 - clamp(0.3 * (1 - 0), 0, 1)
        = 1 - 0.3
        = 0.7
```
Whereas the expected value is 0.3.

Reply:
Logically what it's doing is calculating the target minimum (`target`), calculating the amount of hp drained to reach that minimum (`1 - target`), and adding back a portion of that amount (`target = target + lenience * (1 - target)`).

I can add comments if you like.